packages/extension-ui/src/components/MnemonicInput.tsx

```diff
@@ -0,0 +1,168 @@
+  setError: (error: string) => void;
+  onChange: (seed: string) => void;
+}
+
+const MnemonicInput = ({
+  address,
+  className,
+  error,
+  genesis,
+  onAccountChange,
+  onChange,
+  path,
+  seed,
+  setAddress,
+  setError,
+  type
+}: Props) => {
+  const [mnemonicWords, setMnemonicWords] = useState<string[]>(Array(12).fill(''));
+  const { t } = useTranslation();
+
+  const _handlePaste = useCallback(
+    (event: React.ClipboardEvent) => {
+      event.preventDefault();
+      const MNEMONIC_LENGTH = 12;
+
+      setMnemonicWords([]);
+      const pastedWords = event.clipboardData.getData('text').trim().split(' ').slice(0, 12);
+      const defaultValue = '';
+      const fillArray = Array(MNEMONIC_LENGTH - pastedWords.length).fill(defaultValue);
+      const result = pastedWords.concat(fillArray);
+
+      setMnemonicWords(result);
+      onChange(result.join(' '));
+    },
+    [onChange, setMnemonicWords]
+  );
+
+  const _handleChange = useCallback(
+    (value: string, index: number) => {
+      if (mnemonicWords.every((word) => word === '')) {
+        return;
+      }
+
+      const newMnemonicWords = [...mnemonicWords];
+
+      newMnemonicWords[index] = value;
+
+      onChange(newMnemonicWords.join(' '));
+      setMnemonicWords(newMnemonicWords.slice(0, 12));
+    },
+    [mnemonicWords, onChange, setMnemonicWords]
+  );
+
+  useEffect(() => {
+    setMnemonicWords(seed ? seed.split(' ') : Array(12).fill(''));
+  }, [seed]);
+
+  useEffect(() => {
+    // No need to validate an empty seed
+    // we have a dedicated error for this
+    if (!seed) {
+      onAccountChange(null);
+
+      return;
+    }
+
+    console.log('seed: ', seed);
+
+    const suri = `${seed || ''}${path || ''}`;
+
+    validateSeed(suri, type)
+      .then((validatedAccount) => {
+        console.log('validatedAccount: ', validatedAccount);
+        setError('');
+        setAddress(validatedAccount.address);
+        onAccountChange(objectSpread<AccountInfo>({}, validatedAccount, { genesis, type }));
+      })
+      .catch(() => {
+        setAddress('');
+        onAccountChange(null);
+        setError(path ? t<string>('Invalid mnemonic seed or derivation path') : t<string>('Invalid secret phrase'));
+      });
+  }, [t, genesis, seed, path, onAccountChange, type, setError, setAddress]);
+
+  return (
+    <div
+      className={className}
+      onPaste={_handlePaste}
+    >
+      <div className='mnemonic-container'>
+        {mnemonicWords.map((word, index) => (
+          <MnemonicPill
+            className='mnemonic-pill'
+            index={index + 1}
+            isError={!!error && !!seed}
+            key={index + 1}
+            name={`input${index}`}
+            onChange={_handleChange}
+            readonly={false}
+            word={mnemonicWords[index]}
+            // eslint-disable-next-line react/jsx-no-bind
+          />
+        ))}
+      </div>
+      {!!error && !!seed && <Warning isDanger>{error}</Warning>}
```

Comment:
You are using !!error && seed two times in this file - you could assign it to a const and then use this const in both cases. A bit DRYier and easier to reason about :)